### PR TITLE
SQL: hide system fields instead of drop [closes #210]

### LINKED ISSUE #210: Hide system fields and PERIOD when dropped
```sql
create table t (
  a int,
  sys_trx_start timestamp(6) generated always as row start,
  sys_trx_end timestamp(6) generated always as row end,
  period for system_time (sys_trx_start, sys_trx_end)
) with system versioning;
select * from t;
a	sys_trx_start	sys_trx_end
```
Current behavior:
```sql
alter table t drop sys_trx_start;
query 'alter table t drop sys_trx_start' failed: 4072: Wrong parameters for `t`: Can not drop system versioning field`
```
Required behavior:
```sql
alter table t drop sys_trx_start;
select * from t;
a
```

### PR DESCRIPTION
**Pull request by kevgs**
Actually, does not change `SHOW CREATE TABLE`.

### CODE REVIEW

File: sql/handler.cc [hunk 1]
```diff
@@ -6811,6 +6812,23 @@ static bool add_field_to_drop_list(THD *thd, Alter_info *alter_info,
...
+  if (f->flags & HIDDEN_FLAG)
+  {
+    my_printf_error(ER_VERS_WRONG_PARAMS, "'%s' field aready HIDDEN", MYF(0),
+                    name);
+    return true;
+  }
+  if (vers_create_sys_field(thd, name, alter_info,
+                            sys_start ? VERS_SYS_START_FLAG : VERS_SYS_END_FLAG,
+                            integer_fields))
+    return true;
```
Comment (midenok) [inline sql/handler.cc]: Unreadable formatting: either use single-line **if**, or use braces. Too much new-lines.

File: sql/handler.cc [hunk 2]
```diff
@@ -6872,14 +6890,15 @@ bool Vers_parse_info::check_and_fix_alter(THD *thd, Alter_info *alter_info,
...
+      while (Alter_drop *d= it++)
       {
-        if (is_trx_start(d->name) || is_trx_end(d->name))
-        {
-          my_error(ER_VERS_WRONG_PARAMS, MYF(0), table_name,
-                   "Can not drop system versioning field");
+        const char *name= d->name;
+        if (is_trx_start(name) &&
+            make_hidden(thd, alter_info, integer_fields, name, share, true))
+          return true;
```
Comment (midenok) [inline sql/handler.cc]: Same here.  Remove needless new-lines.

File: sql/handler.cc [hunk 3]
```diff
@@ -6811,6 +6812,23 @@ static bool add_field_to_drop_list(THD *thd, Alter_info *alter_info,
...
 }
 
+static bool make_hidden(THD *thd, Alter_info *alter_info, bool integer_fields,
+                        const char *name, TABLE_SHARE *s, bool sys_start)
+{
+  Field *f= sys_start ? s->vers_start_field() : s->vers_end_field();
+  if (f->flags & HIDDEN_FLAG)
+  {
+    my_printf_error(ER_VERS_WRONG_PARAMS, "'%s' field aready HIDDEN", MYF(0),
+                    name);
```
Comment (midenok) [inline sql/handler.cc]: Seriously? New-line just for `name`?

**Comment by midenok** [comment]
Fix commit message.

**Comment by kevgs** [comment]
> Fix commit message.

Copy PR name or use another one when 'Squash and merge'.

File: sql/handler.cc [hunk 4]
```diff
@@ -6872,12 +6892,17 @@ bool Vers_parse_info::check_and_fix_alter(THD *thd, Alter_info *alter_info,
     if (alter_info->drop_list.elements)
     {
       List_iterator<Alter_drop> it(alter_info->drop_list);
-      while (Alter_drop* d= it++)
+      while (Alter_drop *d= it++)
       {
-        if (is_trx_start(d->name) || is_trx_end(d->name))
+        const char *name= d->name;
+        if (is_trx_start(name) &&
+            make_hidden(thd, alter_info, integer_fields, name, share, true))
```
Comment (midenok) [inline sql/handler.cc]: `make_hidden()` is just another call, so don't spawn stubs. Use:
```c++
Field *f= NULL;
if (!done_start && is_trx_start(d->name))
{
  f= s->vers_start_field();
  done_start= true;
}
else if (!done_end && is_trx_end(d->name))
{
  f= s->vers_end_field();
  done_end= true;
}
else
  continue;
if (f->flags & HIDDEN_FLAG)
{
  // put real DROP error here (same when field is really missing).
}

// call vers_create_sys_field()

if (done_start && done_end)
  break;
```

**Review by midenok — CHANGES_REQUESTED** [review]

File: sql/handler.cc [hunk 5]
```diff
@@ -6891,20 +6872,41 @@ bool Vers_parse_info::check_and_fix_alter(THD *thd, Alter_info *alter_info,
...
+        if (f->flags & HIDDEN_FLAG)
+        {
+          my_error(ER_CANT_DROP_FIELD_OR_KEY, MYF(0), d->type_name(), name);
           return true;
         }
-        if (is_trx_end(name) &&
-            make_hidden(thd, alter_info, integer_fields, name, share, false))
+
+        if (vers_create_sys_field(thd, name, alter_info,
+                                  is_trx_start(name) ? VERS_SYS_START_FLAG
```
Comment (midenok) [inline sql/handler.cc]: You already done `is_trx_start()` check. Why are you doing it again?

**Review by midenok — APPROVED** [review]

**Pull request merged**